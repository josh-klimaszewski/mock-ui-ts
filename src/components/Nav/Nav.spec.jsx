import * as enzyme from 'enzyme';<u></u>
import React from 'react';
import renderer from 'react-test-renderer';
import Nav from './Nav';

describe('Nav bar', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = enzyme.shallow(<Nav />)
        jest.mock('react-router-dom/Link', () => 'Link')
    })
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('calls setActiveNav when a nav item is clicked', () => {
        const spy = jest.spyOn(wrapper.instance(), 'setActiveNav');
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.find('Link').first().simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
    })
    it('updates state.activeNav', () => {
        wrapper.instance().setActiveNav('dashboard');
        expect(wrapper.state('activeNav')).toEqual('dashboard');
    })
})

