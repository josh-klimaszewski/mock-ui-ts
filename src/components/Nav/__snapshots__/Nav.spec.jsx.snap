// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="links"
      >
        <Link
          className="active"
          onClick={[Function]}
          replace={false}
          to="/"
        >
          <i
            className="material-icons md-dark"
          >
            dashboard
          </i>
          Dashboard
        </Link>
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="/invoices"
        >
          <i
            className="material-icons md-dark"
          >
            receipt
          </i>
          Invoices
        </Link>
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="/paybills"
        >
          <i
            className="material-icons md-dark"
          >
            credit_card
          </i>
          Pay Bills
        </Link>
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="/accounting"
        >
          <i
            className="material-icons md-dark"
          >
            assessment
          </i>
          Accounting
        </Link>
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="/reports"
        >
          <i
            className="material-icons md-dark"
          >
            subject
          </i>
          Reports
        </Link>
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="connectedapps/"
        >
          <i
            className="material-icons md-dark"
          >
            device_hub
          </i>
          Connected Apps
        </Link>
        <figure
          className="gap"
        />
        <figure
          className="gap"
        />
        <Link
          className=""
          onClick={[Function]}
          replace={false}
          to="/settings"
        >
          <i
            className="material-icons md-dark"
          >
            settings
          </i>
          Settings
        </Link>
      </div>,
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="active"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            <i
              className="material-icons md-dark"
            >
              dashboard
            </i>
            Dashboard
          </Link>,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/invoices"
          >
            <i
              className="material-icons md-dark"
            >
              receipt
            </i>
            Invoices
          </Link>,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/paybills"
          >
            <i
              className="material-icons md-dark"
            >
              credit_card
            </i>
            Pay Bills
          </Link>,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/accounting"
          >
            <i
              className="material-icons md-dark"
            >
              assessment
            </i>
            Accounting
          </Link>,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/reports"
          >
            <i
              className="material-icons md-dark"
            >
              subject
            </i>
            Reports
          </Link>,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="connectedapps/"
          >
            <i
              className="material-icons md-dark"
            >
              device_hub
            </i>
            Connected Apps
          </Link>,
          <figure
            className="gap"
          />,
          <figure
            className="gap"
          />,
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/settings"
          >
            <i
              className="material-icons md-dark"
            >
              settings
            </i>
            Settings
          </Link>,
        ],
        "className": "links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                dashboard
              </i>,
              "Dashboard",
            ],
            "className": "active",
            "onClick": [Function],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dashboard",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "dashboard",
              "type": "i",
            },
            "Dashboard",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                receipt
              </i>,
              "Invoices",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "/invoices",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "receipt",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "receipt",
              "type": "i",
            },
            "Invoices",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                credit_card
              </i>,
              "Pay Bills",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "/paybills",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "credit_card",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "credit_card",
              "type": "i",
            },
            "Pay Bills",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                assessment
              </i>,
              "Accounting",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "/accounting",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "assessment",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "assessment",
              "type": "i",
            },
            "Accounting",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                subject
              </i>,
              "Reports",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "/reports",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "subject",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "subject",
              "type": "i",
            },
            "Reports",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                device_hub
              </i>,
              "Connected Apps",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "connectedapps/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "device_hub",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "device_hub",
              "type": "i",
            },
            "Connected Apps",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "gap",
          },
          "ref": null,
          "rendered": null,
          "type": "figure",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "gap",
          },
          "ref": null,
          "rendered": null,
          "type": "figure",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons md-dark"
              >
                settings
              </i>,
              "Settings",
            ],
            "className": "",
            "onClick": [Function],
            "replace": false,
            "to": "/settings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "settings",
                "className": "material-icons md-dark",
              },
              "ref": null,
              "rendered": "settings",
              "type": "i",
            },
            "Settings",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="links"
        >
          <Link
            className="active"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            <i
              className="material-icons md-dark"
            >
              dashboard
            </i>
            Dashboard
          </Link>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/invoices"
          >
            <i
              className="material-icons md-dark"
            >
              receipt
            </i>
            Invoices
          </Link>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/paybills"
          >
            <i
              className="material-icons md-dark"
            >
              credit_card
            </i>
            Pay Bills
          </Link>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/accounting"
          >
            <i
              className="material-icons md-dark"
            >
              assessment
            </i>
            Accounting
          </Link>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/reports"
          >
            <i
              className="material-icons md-dark"
            >
              subject
            </i>
            Reports
          </Link>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="connectedapps/"
          >
            <i
              className="material-icons md-dark"
            >
              device_hub
            </i>
            Connected Apps
          </Link>
          <figure
            className="gap"
          />
          <figure
            className="gap"
          />
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/settings"
          >
            <i
              className="material-icons md-dark"
            >
              settings
            </i>
            Settings
          </Link>
        </div>,
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="active"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              <i
                className="material-icons md-dark"
              >
                dashboard
              </i>
              Dashboard
            </Link>,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/invoices"
            >
              <i
                className="material-icons md-dark"
              >
                receipt
              </i>
              Invoices
            </Link>,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/paybills"
            >
              <i
                className="material-icons md-dark"
              >
                credit_card
              </i>
              Pay Bills
            </Link>,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/accounting"
            >
              <i
                className="material-icons md-dark"
              >
                assessment
              </i>
              Accounting
            </Link>,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/reports"
            >
              <i
                className="material-icons md-dark"
              >
                subject
              </i>
              Reports
            </Link>,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="connectedapps/"
            >
              <i
                className="material-icons md-dark"
              >
                device_hub
              </i>
              Connected Apps
            </Link>,
            <figure
              className="gap"
            />,
            <figure
              className="gap"
            />,
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/settings"
            >
              <i
                className="material-icons md-dark"
              >
                settings
              </i>
              Settings
            </Link>,
          ],
          "className": "links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  dashboard
                </i>,
                "Dashboard",
              ],
              "className": "active",
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dashboard",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "dashboard",
                "type": "i",
              },
              "Dashboard",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  receipt
                </i>,
                "Invoices",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/invoices",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "receipt",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "receipt",
                "type": "i",
              },
              "Invoices",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  credit_card
                </i>,
                "Pay Bills",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/paybills",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "credit_card",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "credit_card",
                "type": "i",
              },
              "Pay Bills",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  assessment
                </i>,
                "Accounting",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/accounting",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "assessment",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "assessment",
                "type": "i",
              },
              "Accounting",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  subject
                </i>,
                "Reports",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/reports",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "subject",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "subject",
                "type": "i",
              },
              "Reports",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  device_hub
                </i>,
                "Connected Apps",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "connectedapps/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "device_hub",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "device_hub",
                "type": "i",
              },
              "Connected Apps",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "gap",
            },
            "ref": null,
            "rendered": null,
            "type": "figure",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "gap",
            },
            "ref": null,
            "rendered": null,
            "type": "figure",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons md-dark"
                >
                  settings
                </i>,
                "Settings",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/settings",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "settings",
                  "className": "material-icons md-dark",
                },
                "ref": null,
                "rendered": "settings",
                "type": "i",
              },
              "Settings",
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
